### PR TITLE
New: Auto-generated captions

### PR DESCRIPTION
**Pull request by tonyjin**
Note: This depends on video intelligence being enabled for your enterprise. See https://www.box.com/skills for details.

This patch detects if Box Skills transcript cards are loaded and if so, converts them to Shaka Text Cues to be used for captioning. This patch doesn't support switching between native subtitles and auto-generated subtitles. If there are subtitles already within the video, we do not attempt to load auto-generated captions.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tonyjin has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/media/DashViewer.js [hunk 1]
```diff
@@ -279,12 +283,26 @@ class DashViewer extends VideoBaseViewer {
...
         const subtitleIdx = parseInt(this.cache.get('media-subtitles'), 10);
-        if (this.textTracks[subtitleIdx] !== undefined) {
+
+        // Auto-generated index 0 ==> turn auto-generated text track on
+        if (this.autoCaptionDisplayer && subtitleIdx === 0) {
+            // Manually set text visibility with the custom Shaka Text Displayer
+            this.autoCaptionDisplayer.setTextVisibility(true);
+            this.emit('subtitlechange', __('auto_generated'));
+
+            // Valid non-auto-generated index ==> turn specified text track on
```
Comment (jeremypress) [inline src/lib/viewers/media/DashViewer.js]: Move these comments into their blocks?

File: src/lib/viewers/media/DashViewer.js [hunk 2]
```diff
@@ -426,13 +444,56 @@ class DashViewer extends VideoBaseViewer {
...
      */
     loadSubtitles() {
+        // Load subtitles from video, if available
         this.textTracks = this.player.getTextTracks().sort((track1, track2) => track1.id - track2.id);
         if (this.textTracks.length > 0) {
             this.mediaControls.initSubtitles(
                 this.textTracks.map((track) => getLanguageName(track.language) || track.language),
                 getLanguageName(this.options.location.locale.substring(0, 2))
             );
+            return;
```
Comment (jeremypress) [inline src/lib/viewers/media/DashViewer.js]: So the behavior for now is only try to load autogenerated captions if there are no baked in subtitles on the video?

File: src/lib/viewers/media/DashViewer.js [hunk 3]
```diff
@@ -426,13 +444,56 @@ class DashViewer extends VideoBaseViewer {
...
+        // @TODO 07-07-18: Support both auto-generated captions and subtitles from videos
+        this.loadAutoGeneratedCaptions();
+    }
+
+    /**
+     * Loads auto-generated captions
+     *
+     * @return {boolean} Whether or not there are any auto-generated captions
+     */
+    loadAutoGeneratedCaptions() {
```
Comment (jeremypress) [inline src/lib/viewers/media/DashViewer.js]: Why does this need to return a boolean?

File: src/lib/viewers/media/DashViewer.js (hunk 1 above)
Comment (tonyjin) [reply src/lib/viewers/media/DashViewer.js]: I like having the comment above the if statement to describe the condition going into the block as well

**Pull request merged**